src/server/scripts/EasternKingdoms/zone_silverpine_forest.cpp

```diff
@@ -0,0 +1,945 @@
+                {
+                    if (_transformDone)
+                        return;
+
+                    DoCastSelf(SPELL_FORSAKEN_TROOPER_MS_COMETH);
+
+                    _transformDone = true;
+
+                    _events.ScheduleEvent(EVENT_FACE_TOWARDS_SYLVANAS, 1s + 500ms);
+                    break;
+                }
+
+                case EVENT_FACE_TOWARDS_SYLVANAS:
+                    me->SetFacingTo(0.706837f);
+                    _events.ScheduleEvent(EVENT_EMOTE_TO_SYLVANAS, 2s + 500ms);
+                    break;
+
+                case EVENT_EMOTE_TO_SYLVANAS:
+                    me->HandleEmoteCommand(EMOTE_ONESHOT_SALUTE);
+                    me->DespawnOrUnsummon(80s);
+                    break;
+
+                default:
+                    break;
+            }
+        }
+    }
+
+private:
+    EventMap _events;
+    bool _transformDone;
+};
+
+enum SpellForsakenTrooperMasterScriptCometh
+{
+    SPELL_FORSAKEN_TROOPER_MALE_01_HC       = 83150,
+    SPELL_FORSAKEN_TROOPER_MALE_02_HC       = 83163,
+    SPELL_FORSAKEN_TROOPER_MALE_03_HC       = 83164,
+    SPELL_FORSAKEN_TROOPER_MALE_04_HC       = 83165,
+    SPELL_FORSAKEN_TROOPER_FEMALE_01_HC     = 83152,
+    SPELL_FORSAKEN_TROOPER_FEMALE_02_HC     = 83166,
+    SPELL_FORSAKEN_TROOPER_FEMALE_03_HC     = 83167,
+    SPELL_FORSAKEN_TROOPER_FEMALE_04_HC     = 83168,
+
+    DISPLAY_MALE_01_HC                      = 33978,
+    DISPLAY_MALE_02_HC                      = 33980,
+    DISPLAY_MALE_03_HC                      = 33979,
+    DISPLAY_MALE_04_HC                      = 33981,
+    DISPLAY_FEMALE_01_HC                    = 33982,
+    DISPLAY_FEMALE_02_HC                    = 33983,
+    DISPLAY_FEMALE_03_HC                    = 33984,
+    DISPLAY_FEMALE_04_HC                    = 33985
+};
+
+// Forsaken Trooper Master Script (Forsaken High Command) - 83149
+class spell_silverpine_forsaken_trooper_masterscript_high_command : public SpellScript
+{
+    PrepareSpellScript(spell_silverpine_forsaken_trooper_masterscript_high_command);
+
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo
+        ({
+            SPELL_FORSAKEN_TROOPER_MALE_01_HC,
+            SPELL_FORSAKEN_TROOPER_MALE_02_HC,
+            SPELL_FORSAKEN_TROOPER_MALE_03_HC,
+            SPELL_FORSAKEN_TROOPER_MALE_04_HC,
+            SPELL_FORSAKEN_TROOPER_FEMALE_01_HC,
+            SPELL_FORSAKEN_TROOPER_FEMALE_02_HC,
+            SPELL_FORSAKEN_TROOPER_FEMALE_03_HC,
+            SPELL_FORSAKEN_TROOPER_FEMALE_04_HC
+        });
+    }
+
+    void HandleScriptEffect(SpellEffIndex /*effIndex*/)
+    {
+        if (Unit* unit = GetCaster())
+        {
+            unit->RemoveAura(SPELL_FEIGNED);
+
+            if (unit->GetDisplayId() == DISPLAY_MALE_01_HC)
+                unit->CastSpell(unit, SPELL_FORSAKEN_TROOPER_MALE_01_HC, true);
+
+            if (unit->GetDisplayId() == DISPLAY_MALE_02_HC)
```

Comment:
either use else ifs here, a switch-case or a static const map

also only do `unit->CastSpell` once, just set a local variable `spellId` if a related condition is satisfied

Reply:
feedback applied.